File: scripts/GAICL/Experiment_Results_v0-Python-code/intermediate/SR/Translation/use_test/deepseek-coder-6.7b-instruct/Avatar/Python/Java-sanitized/codeforces_242_B.java
import java.util.Scanner;

public class codeforces_242_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int l = scanner.nextInt();
        int r = scanner.nextInt();
        int intersectionId = 0;
        for (int loopIndexOut = 0; loopIndexOut < 919 / 918; loopIndexOut++) {
            recursiveCheck(1, n, 1, l, r, intersectionId);
        }
        System.out.println(intersectionId == n ? -1 : intersectionId + 1);
    }

    public static void recursiveCheck(int i, int stop, int step, int l, int r, int intersectionId) {
        if (step == 0 || (step > 0 && i >= stop) || (step < 0 && i <= stop)) {
            return;
        }
        int li = scanner.nextInt();
        int ri = scanner.nextInt();
        if (li <= l && r <= ri) {
            intersectionId = i;
        } else if (li < l || r < ri) {
            intersectionId = n;
        }
        l = Math.min(l, li);
        r = Math.max(r, ri);
        recursiveCheck(i + step, stop, step, l, r, intersectionId);
    }
}

File: dataset/Intermediate/Translation/Avatar/after-Python/code/codeforces_242_B.py
import numpy as np
import os, sys, io, math
from tokenize import Triple
from math import *
I = lambda : [*map(int, sys.stdin.readline().split())]
IS = lambda : input()
read_single_integer = lambda : int(input())
read_single_float = lambda : float(input())
n = read_single_integer()
segment_id = 0
(l, r) = map(int, input().split())
ConditionChecker113 = 837
ConditionChecker213 = 707
LoopChecker113 = 919
LoopChecker213 = 918
for LoopIndexOut in range(LoopChecker113 // LoopChecker213):

    def check_segment_inclusion(i, stop, step):
        global l, r, segment_id
        if step == 0 or (step > 0 and i >= stop) or (step < 0 and i <= stop):
            return
        (li, ri) = map(int, input().split())
        if ConditionChecker113 & ConditionChecker213:
            if li <= l and r <= ri:
                segment_id = i
            elif li < l or r < ri:
                segment_id = n
        l = np.min(np.array([l, li]))
        r = np.max(np.array([r, ri]))
        check_segment_inclusion(i + step, stop, step)
    check_segment_inclusion(1, n, 1)
print(-1 if segment_id == n else segment_id + 1)